onnxruntime/core/platform/windows/env.cc

```diff
@@ -14,23 +14,32 @@ limitations under the License.
 ==============================================================================*/
 // Portions Copyright (c) Microsoft Corporation
 
+#include "core/platform/env.h"
+
 #include <Shlwapi.h>
 #include <Windows.h>
 
+#include <fstream>
 #include <string>
 #include <thread>
+
 #include <fcntl.h>
-#include <fstream>
 #include <io.h>
 
 #include "core/common/logging/logging.h"
-#include "core/platform/env.h"
+#include "core/platform/scoped_resource.h"
 
 namespace onnxruntime {
 
 namespace {
 
-static void DeleteBuffer(void* param) noexcept { ::free(param); }
+struct FileHandleTraits {
+  using Handle = HANDLE;
+  static Handle GetInvalidHandleValue() noexcept { return INVALID_HANDLE_VALUE; }
+  static void CleanUp(Handle h) noexcept { CloseHandle(h); }
+};
+
+using ScopedFileHandle = ScopedResource<FileHandleTraits>;
 
```

Comment:
Please add a comment here: the file should be read only, so that CloseHandle will never fail.  That's the main reason why I didn't use RAII in the original code.

Reply:
http://man7.org/linux/man-pages/man2/close.2.html

A careful programmer will check the return value of close(), since it is quite possible that errors on a previous write(2) operation are reported only on the final close() that releases the open file description.  If you ignore the error, you may loss data.  And in production env, disk failure is pretty common, it happens every day.